tests/console/kdump_and_crash.pm

```diff
@@ -46,10 +46,18 @@ sub run {
     # all but PPC64LE arch's vmlinux images are gzipped
     my $suffix = get_var('OFW') ? '' : '.gz';
     assert_script_run 'find /var/crash/';
+
     my $crash_cmd = "echo exit | crash `ls -1t /var/crash/*/vmcore | head -n1` /boot/vmlinux-`uname -r`$suffix";
-    assert_script_run "$crash_cmd", 600;
-    validate_script_output "$crash_cmd", sub { m/PANIC/ }, 600;
+    if (script_run($crash_cmd)) {
+        record_soft_failure 'boo#1090659 - crash: invalid kernel virtual address: xxxxxxxxx  type: yyyyyyyy';
+        script_run 'ls -lah /boot/';
+        script_run 'tar -cvf /tmp/crash_saved.tar /var/crash/*';
+        upload_logs '/tmp/crash_saved.tar' if is_sle('15+');
+    }
+    else {
+        assert_script_run "$crash_cmd", 600;
+        validate_script_output "$crash_cmd", sub { m/PANIC/ }, 600;
```

Comment:
Already in the original code the `$crash_cmd` was called twice? Would it work to call `script_run "$crash_cmd |& tee /dev/$serialdev` and then check with `wait_serial`? I guess not in this order as the serial output is consumed by script_run checking for the exit status. But probably `script_run '…', 0` and an explicit wait_serial for both the output *and* the exit code would work if you want to play around with that, e.g. something like

```perl
script_run("$crash_cmd |& tee /dev/$serialdev; echo crash-status-\$? > /dev/$serialdev", 0);
if (!wait_serial('PANIC', 600)) {
    diag "crash could not find crash file";
    …
    return;
}
wait_serial('crash-status-0') or die "crash did not complete successfully";
```